src/toil/test/src/fileStoreTest.py

```diff
@@ -233,6 +235,97 @@ def _testFileStoreOperations(job, nonLocalDir, numIters=100):
                             logger.info('No longer have file: %s', fsID)
                 i += 1
 
+        def testWriteReadGlobalFilePermissions(self):
+            """
+            Ensures that uploaded files preserve their file permissions when they
+            are downloaded again. This function checks that a written executable file
+            maintains its executability after being read.
+            """
+            for executable in True,False:
+                A = Job.wrapJobFn(self._testWriteReadGlobalFilePermissions, executable=executable)
+                Job.Runner.startToil(A, self.options)
+
+        @staticmethod
+        def _testWriteReadGlobalFilePermissions(job, executable):
+            workDir = job.fileStore.getLocalTempDir()
+            srcFile = f'{workDir}/in{uuid4()}'
+            with open(srcFile, 'w') as f:
+                f.write('Hello')
+
+            if executable:
+                os.chmod(srcFile, os.stat(srcFile).st_mode | stat.S_IXUSR)
+
+            # Initial file owner execute permissions
+            initialPermissions = os.stat(srcFile).st_mode & stat.S_IXUSR
+            fileID = job.fileStore.writeGlobalFile(srcFile)
+
+            for mutable in True,False:
+                for symlink in True,False:
+                    #with self.subTest(f'Now testing readGlobalFileWith: mutable={mutable} symlink={symlink}'):
+                    dstFile = job.fileStore.getLocalTempFile() + str(uuid4())
```

Comment:
If you just want a filename in a temporary directory, you want to use [`getLocalTempDir()`](https://toil.readthedocs.io/en/releases-5.1.0/developingWorkflows/toilAPIFilestore.html#toil.fileStores.abstractFileStore.AbstractFileStore.getLocalTempDir) or [`getLocalTempFileName()`](https://toil.readthedocs.io/en/releases-5.1.0/developingWorkflows/toilAPIFilestore.html#toil.fileStores.abstractFileStore.AbstractFileStore.getLocalTempFileName) and not [`getLocalTempFile()`](https://toil.readthedocs.io/en/releases-5.1.0/developingWorkflows/toilAPIFilestore.html#toil.fileStores.abstractFileStore.AbstractFileStore.getLocalTempFile) which creates a file for you.

Reply:
This is still the version that creates an extraneous file. But it should still *work*, so it might be good enough.